### PR TITLE
return empty bytes (b'' ) for h5py.Empty strings

### LINKED ISSUE #94: Inconsistent attribute handling between netcdf4 and h5netcdf
I have found a netCDF file that cannot be decoded by xarray via the h5netcdf engine but CAN be decoded via netCDF4. 

This file will reproduce the bug
```
! wget 'https://esgf-world.s3.amazonaws.com/CMIP6/CMIP/IPSL/IPSL-CM6A-LR/abrupt-4xCO2/r1i1p1f1/Lmon/cLeaf/gr/v20190118/cLeaf_Lmon_IPSL-CM6A-LR_abrupt-4xCO2_r1i1p1f1_gr_185001-214912.nc'
```

```python
import netCDF4
import h5netcdf.legacyapi as netCDF4_h5

local_path = "cLeaf_Lmon_IPSL-CM6A-LR_abrupt-4xCO2_r1i1p1f1_gr_185001-214912.nc"
with netCDF4_h5.Dataset(local_path, mode='r') as ncfile:
    print('h5netcdf:', ncfile['cLeaf'].getncattr("coordinates"))
with netCDF4.Dataset(local_path, mode='r') as ncfile:
    #assert "coordinates" not in ncfile['cLeaf'].attrs
    print('netCDF4:', ncfile['cLeaf'].getncattr("coordinates"))
```

```
h5netcdf: Empty(dtype=dtype('S1'))
netCDF4: 
```

As we can see, we get an empty string `''` in netCDF4 but a `<class 'h5py._hl.base.Empty'>` object from h5netcdf. This weird attribute prevents xarray from decoding the dataset.

**Environment**:

<details><summary>Output of <tt>xr.show_versions()</tt></summary>

INSTALLED VERSIONS
------------------
commit: None
python: 3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:22:27) 
[GCC 9.3.0]
python-bits: 64
OS: Linux
OS-release: 4.19.150+
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: C.UTF-8
LANG: C.UTF-8
LOCALE: en_US.UTF-8
libhdf5: 1.10.6
libnetcdf: 4.7.4

xarray: 0.17.0
pandas: 1.2.3
numpy: 1.20.2
scipy: 1.6.2
netCDF4: 1.5.6
pydap: installed
h5netcdf: 0.10.0
h5py: 3.1.0
Nio: None
zarr: 2.7.0
cftime: 1.4.1
nc_time_axis: 1.2.0
PseudoNetCDF: None
rasterio: 1.2.1
cfgrib: 0.9.8.5
iris: None
bottleneck: 1.3.2
dask: 2021.03.1
distributed: 2021.03.1
matplotlib: 3.3.4
cartopy: 0.18.0
seaborn: None
numbagg: None
pint: 0.17
setuptools: 49.6.0.post20210108
pip: 20.3.4
conda: None
pytest: None
IPython: 7.22.0
sphinx: None

</details>

xref https://github.com/pangeo-forge/pangeo-forge/issues/105, https://github.com/pydata/xarray/issues/5172

### PR DESCRIPTION
**Pull request by kmuehlbauer**
closes #94

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@rabernat According to @shoyer's suggestions this should close the case. 

If `h5py.Empty` strings are detected they are extracted as empty bytes (`b''`).

File: h5netcdf/attrs.py [hunk 1]
```diff
@@ -22,8 +22,14 @@ def __init__(self, h5attrs, check_dtype):
         self._check_dtype = check_dtype
 
     def __getitem__(self, key):
+        import h5py
+
         if key in _HIDDEN_ATTRS:
             raise KeyError(key)
+        if isinstance(self._h5attrs[key], h5py.Empty):
```
Comment (shoyer) [inline h5netcdf/attrs.py:29]: Maybe add a comment referencing the issue where I explained what is going on down?

**Review by shoyer — APPROVED** [review]

File: h5netcdf/attrs.py (hunk 1 above)
Comment (kmuehlbauer) [reply h5netcdf/attrs.py:30]: Done!

**Comment by kmuehlbauer** [comment]
I've rebased on latest master and added a changelog entry. Will merge, when CI is happy.

**Pull request merged**